woocommerce/wc-admin
# Components: Add filter picker

This PR adds the Filter Picker to a Products Report. The filter picker accepts a configuration array and is capable of supporting multiple levels of subFilters.

**Designs** p6riRB-3lk-p2

### For the next PRs
* Animations
* Search component for selecting a single product

### Screeshots
Closed
![screen shot 2018-07-16 at 4 17 34 pm](https://user-images.githubusercontent.com/1922453/42743022-410d9846-8914-11e8-8a69-2f4e0841daec.png)

Open
![screen shot 2018-07-16 at 4 17 43 pm](https://user-images.githubusercontent.com/1922453/42743025-4d5ca3ee-8914-11e8-9548-bf93f6d49898.png)

Search Component (to be created in following PR)
![screen shot 2018-07-16 at 4 18 02 pm](https://user-images.githubusercontent.com/1922453/42743031-53c6c980-8914-11e8-9b34-55a074ac2616.png)

### Testing

1. Go to `/wp-admin/admin.php?page=wc-admin#/analytics/products`
2. Make selections and see the url parameter update (except "Single Product")
3. Add other parameters by using the datepicker and ensure the Filter Picker doesn't interfere with those
4. Make sure accessibility via keyboard works as expected
5. Make sure the selection is correctly referenced in the UI. "All Products" is the default
6. Compare the styles (selected, hover, focus) with datepicker selection and note any differences
7. Test different screen widths to make sure the Datepicker and Filterpicker are rendered correctly



Fixes https://github.com/woocommerce/wc-admin/issues/191
